### PR TITLE
Phase 1.5 (B2C) in the tutorial 

### PR DESCRIPTION
**Pull request by TiagoBrenck**
## Purpose
Issue: https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/issues/112
Phase 1.5 to the tutorial to show how to sign-in users with social identities or local identities with Azure AD B2C

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Get the code
* Follow the Readme.md steps to create your Azure B2C

## What to Check
Verify that the following are valid
* Test if you can sign-up then sign-in

### CODE REVIEW

File: 1-WebApp-OIDC/1-5-B2C/README.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+### Step 1: Clone or download this repository
+
+From your shell or command line:
+
+```powershell
+git clone https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2.git
+```
+
+> Given that the name of the sample is very long, and so are the name of the referenced NuGet packages, you might want to clone it in a folder close to the root of your hard drive, to avoid file size limitations on Windows.
```
Comment (jennyf19) [inline 1-WebApp-OIDC/1-5-B2C/README.md]: >name [](start = 65, length = 4)

names

File: 1-WebApp-OIDC/1-5-B2C/README.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+### Step 2: Get your own Azure AD B2C tenant
+
+If you don't an Azure AD B2C tenant yet, you'll need to create an Azure AD B2C tenant by following [these instructions](https://azure.microsoft.com/documentation/articles/active-directory-b2c-get-started).
```
Comment (jennyf19) [inline 1-WebApp-OIDC/1-5-B2C/README.md]: have

Comment (jennyf19) [inline 1-WebApp-OIDC/1-5-B2C/README.md]: >instructions [](start = 106, length = 12)

nit: consider expanding on what is happening here, so they can decide to click the link or not..ex "...tenant by following [these instructions on creating an AAD B2C tenant]()

File: 1-WebApp-OIDC/1-5-B2C/README.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+### Step 3: Create your own user flow(policy)
```
Comment (jennyf19) [inline 1-WebApp-OIDC/1-5-B2C/README.md]: nit: space between flow and (policy)

File: 1-WebApp-OIDC/1-5-B2C/README.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+- Enable the **Web App/Web API** setting for your application.
+- Set the **Reply URL** to `https://localhost:44321//signin/B2C_1_sign_up_in` and `https://localhost:44321//signout/B2C_1_sign_up_in`.
+- Copy the Application ID generated for your application, so you can use it in the next step.
+
+### Step 5: Configure the sample with your app coordinates
+
+1. Open the solution in Visual Studio.
+1. Open the `appsettings.json` file.
+1. Find the assignment for `Instance` and replace the value with `<your-tenant-name>.b2clogin.com`
```
Comment (jennyf19) [inline 1-WebApp-OIDC/1-5-B2C/README.md]: ><your-tenant-name>.b2clogin.com [](start = 66, length = 31)

this isn't very clear...maybe something like "replace the value of <your-tenant-name> with your tenant name" and give an example?

File: 1-WebApp-OIDC/1-5-B2C/Startup.cs [hunk 5]
```diff
@@ -1,10 +1,8 @@
-﻿using Microsoft.AspNetCore.Authentication.AzureAD.UI;
-using Microsoft.AspNetCore.Authorization;
+﻿using Microsoft.AspNetCore.Authentication.AzureADB2C.UI;
 using Microsoft.AspNetCore.Builder;
```
Comment (jennyf19) [inline 1-WebApp-OIDC/1-5-B2C/Startup.cs]: license info?

**Comment by jennyf19** [comment]
> using Microsoft.AspNetCore.Authentication;

license info?

---
Refers to: 1-WebApp-OIDC/1-5-B2C/Controllers/AuthController.cs:1 in 08520ef. [](commit_id = 08520ef6da4643a1c714f5d66af33b48ce834f72, deletion_comment = False)

File: 1-WebApp-OIDC/1-5-B2C/Views/Home/Index.cshtml [hunk 6]
```diff
@@ -3,10 +3,10 @@
 }
 
 <h1>
-    ASP.NET Core web app signing-in users in your organization
+    ASP.NET Core web app signing-in users in B2C
```
Comment (jennyf19) [inline 1-WebApp-OIDC/1-5-B2C/Views/Home/Index.cshtml]: Add "Azure AD B2C"?

File: 1-WebApp-OIDC/1-5-B2C/Views/Home/Index.cshtml [hunk 7]
```diff
@@ -3,10 +3,10 @@
...
 </h1>
 <p>
-    This sample shows how to build a .NET Core 2.2 MVC Web app that uses OpenID Connect to sign in users in your organization. It leverages the ASP.NET Core OpenID Connect middleware.
+    This sample shows how to build a .NET Core 2.2 MVC Web app that uses OpenID Connect to sign in users in B2C. It leverages the ASP.NET Core OpenID Connect middleware.
```
Comment (jennyf19) [inline 1-WebApp-OIDC/1-5-B2C/Views/Home/Index.cshtml]: >B2C [](start = 108, length = 3)

they are signing in w/a variety of IdPs...so maybe reword this somehow to include that?

File: 1-WebApp-OIDC/1-5-B2C/README.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+---
+# An ASP.NET Core Web app signing-in users with the Microsoft identity platform in Azure AD B2C
+
+[![Build status](https://identitydivision.visualstudio.com/IDDP/_apis/build/status/AAD%20Samples/.NET%20client%20samples/ASP.NET%20Core%20Web%20App%20tutorial)](https://identitydivision.visualstudio.com/IDDP/_build/latest?definitionId=819)
+
+## Scenario
+
+This sample shows how to build a .NET Core 2.2 MVC Web app that uses OpenID Connect to sign in users in **Azure AD B2C**. It assumes you have some familiarity with **Azure AD B2C**. If you'd like to learn all that B2C has to offer, start with our documentation at aka.ms/aadb2c.
+
+![Sign in with Azure AD](ReadmeFiles/sign-in.png)
```
Comment (kalyankrishna1) [inline 1-WebApp-OIDC/1-5-B2C/README.md]: >sign-in.png [](start = 37, length = 11)

This needs updating, should show Azure AD B2C instead of Azure AD

File: Microsoft.Identity.Web/B2CStartupHelper.cs [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        /// <summary>
+        /// Add authentication with Azure Ad B2C.
+        /// This expects the configuration files will have a section named "AzureAdB2C"
+        /// </summary>
+        /// <param name="services">Service collection to which to add this authentication scheme</param>
+        /// <param name="configuration">The Configuration object</param>
+        /// <returns></returns>
+        public static IServiceCollection AddAzureAdB2CAuthentication(this IServiceCollection services, IConfiguration configuration)
+        {
+            var b2cOptions = configuration.GetSection("AzureAdB2C").Get<AzureADB2COptions>();
```
Comment (jmprieur) [inline Microsoft.Identity.Web/B2CStartupHelper.cs]: @TiagoBrenck:  Is there a reason, not to have leveraged `AddAzureADB2C`:
- https://github.com/aspnet/AspNetCore/blob/master/src/Azure/AzureAD/Authentication.AzureADB2C.UI/src/AzureAdB2CAuthenticationBuilderExtensions.cs#L80
or 
- https://github.com/aspnet/AspNetCore/blob/9f52639909df27efa01017e777ca9703dc77a2ab/src/Azure/AzureAD/Authentication.AzureADB2C.UI/src/AzureAdB2CAuthenticationBuilderExtensions.cs#L100

File: 1-WebApp-OIDC/1-5-B2C/Controllers/AuthController.cs [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+using Microsoft.AspNetCore.Authentication.AzureADB2C.UI;
+using Microsoft.AspNetCore.Authentication.Cookies;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WebApp_OpenIDConnect_DotNet.Controllers
+{
+    public class AuthController : Controller
```
Comment (jmprieur) [inline 1-WebApp-OIDC/1-5-B2C/Controllers/AuthController.cs]: @TiagoBrenck is there a reason for adding an auth controller, and not let ASP.NET Core leverage the B2C AccountController.cs
https://github.com/aspnet/AspNetCore/blob/master/src/Azure/AzureAD/Authentication.AzureADB2C.UI/src/Areas/AzureADB2C/Controllers/AccountController.cs
?
